### PR TITLE
Budget project

### PR DESCRIPTION
**Pull request by Theodoraldo**
In this PR, a budget application was built from scratch to display the amount a specific user makes over a period of time. Below describe the full details of the project : 

#### Requirements
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,189 @@
...
+
+- [Live Demo Link](https://budget-ok43.onrender.com)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Walkthrough Demo <a name="live-demo"></a>
+
+- [Walkthrogh Demo Link](https://www.loom.com/share/4a49fbad89c746988d139f5ed4b3900b)
```
Comment (TadesseDev) [inline README.md]: - [ ] Dear @Theodoraldo, good job with the video documentation. I notice The video sound is not available. I could not listen to what you have said. Please be reminded we need to have professional documentation containing all the details with both sound and video. Consider redoing the version adding all details including the audio.

File: app/views/entities/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<h1 class="is-size-4 mb-3 has-text-grey is-flex is-justify-content-center has-text-weight-bold">Transaction Details</h1>
+
+<div class="container" style="width: 90%;">
+<p class="is-size-5 is-italic has-text-centered my-3 has-text-weight-bold has-text-grey"><%= @group.name %></p>
```
Comment (TadesseDev) [inline app/views/entities/show.html.erb:4]: - [ ] Dear @Theodoraldo, According to the project requirements, we need to validate all the input fields form the user. This way we can protect our app form unexpected crashes and can save user from making silly mistakes. currently when I try to register a transaction without feeling the form the application crashes and I get an exception. Instead, we can make the input fields required from the HTML and we can also handle such invalid inputs from the backend and protect our app from crashing. 

Screen Shot 2023-10-19 at 11 44 58 PM

File: README.md [hunk 3]
```diff
@@ -1,2 +1,189 @@
...
+
+> If you like this project give it a ⭐️
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- We would like to thank Microverse and anyone who contributed to the success of this project. If we am to mention names, this readme cannot hold the list 😁😁.
```
Comment (TadesseDev) [inline README.md]: - [ ] Dear @Theodoraldo, Please be reminded the README is expected to contain an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). Pleae consider adding one.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Theodoraldo 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Good job overall following the template design ✅
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 4]
```diff
@@ -1,2 +1,189 @@
...
+
+> If you like this project give it a ⭐️
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+- I would like to acknowledge Gregoire Vella for such an awesome design and documentation.
+- Thanks to Microverse and anyone who contributed to the success of this project. If we am to mention names, this readme cannot hold the list 😁😁.
```
Comment (Felix45) [inline README.md]: - [ Required ] Good job adding this section as requested by the previous reviewer, however, please note you forgot to include the links that would take a user to [Gregoire Vella's](https://www.behance.net/gregoirevella) profile. I would like to suggest that you edit this section as shown below so as to make your acknowledgments section more professional.

## Sample attribution
I would like to acknowledge [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) for such an awesome design and documentation.

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Theodoraldo, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Application functions as expected :white_check_mark: 
- [x] Good presentation video :white_check_mark: 
- [x] Implemented most of the changes requested by the previous reviewer :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark:     

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @Theodoraldo , 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- Good job for adding only necessary routes for your controller's actions :+1: 

## Optional suggestions

- It is always recommendation to follow the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) for any solo project you are building. This project will be deploy directly to the main branch which is not accepted. You may commit some errors and your app will still be deploy automatically. Always create features branch and make the dev branch default for automatic deployment. Open PR from feature to dev branch

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**